custom_components/grocy/entity.py

```diff
@@ -1,108 +1,44 @@
-"""GrocyEntity class"""
+"""Entity for Grocy."""
+from __future__ import annotations
+
 import json
+from collections.abc import Mapping
+from typing import Any
 
 from homeassistant.helpers.device_registry import DeviceEntryType
+from homeassistant.helpers.entity import DeviceInfo
 from homeassistant.helpers.update_coordinator import CoordinatorEntity
 
-# pylint: disable=relative-beyond-top-level
-from .const import (
-    DOMAIN,
-    GrocyEntityIcon,
-    GrocyEntityType,
-    GrocyEntityUnit,
-    NAME,
-    VERSION,
-)
-from .json_encode import GrocyJSONEncoder
-
-
-class GrocyEntity(CoordinatorEntity):
-    """Base class for Grocy entities."""
-
-    def __init__(self, coordinator, config_entry, entity_type):
-        """Initialize generic Grocy entity."""
-        super().__init__(coordinator)
-        self.config_entry = config_entry
-        self.entity_type = entity_type
-
-    @property
-    def unique_id(self):
-        """Return a unique ID to use for this entity."""
-        return f"{self.config_entry.entry_id}{self.entity_type.lower()}"
-
-    @property
-    def name(self):
-        """Return the name of the binary_sensor."""
-        return f"{NAME} {self.entity_type.lower().replace('_', ' ')}"
-
-    @property
-    def entity_registry_enabled_default(self) -> bool:
-        """Return if the entity should be enabled when first added to the entity registry."""
-        return False
+from .const import CONF_URL, DOMAIN, NAME, VERSION
+from .coordinator import GrocyDataUpdateCoordinator
+from .json_encoder import CustomJSONEncoder
 
-    @property
-    def entity_data(self):
-        """Return the entity_data of the entity."""
-        if self.coordinator.data is None:
-            return None
-        return self.coordinator.data.get(self.entity_type)
 
-    @property
-    def unit_of_measurement(self):
-        """Return the unit the value is expressed in."""
-        if GrocyEntityType(self.entity_type).name in [x.name for x in GrocyEntityUnit]:
-            return GrocyEntityUnit[GrocyEntityType(self.entity_type).name]
+class GrocyEntity(CoordinatorEntity[GrocyDataUpdateCoordinator]):
+    """Grocy base entity definition."""
 
     @property
-    def icon(self):
-        """Return the icon of the entity."""
-        if GrocyEntityType(self.entity_type).name in [x.name for x in GrocyEntityIcon]:
-            return GrocyEntityIcon[GrocyEntityType(self.entity_type).name]
-
-        return GrocyEntityIcon.DEFAULT
+    def device_info(self) -> DeviceInfo:
+        """Grocy device information."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, self.coordinator.config_entry.entry_id)},
+            name=NAME,
+            manufacturer=NAME,
+            model=VERSION,  # TODO This should be the Grocy version
```

Comment:
`model` should probably not be used at all?
There are a `software_version` key that fits better.